rust/src/graph/mod.rs

```diff
@@ -0,0 +1,137 @@
+use bimap::BiMap;
+use derive_new::new;
+use getset::{CopyGetters, Getters};
+use lazy_static::lazy_static;
+use rustc_hash::{FxHashMap, FxHashSet};
+use slotmap::{new_key_type, SecondaryMap, SlotMap};
+use std::sync::RwLock;
+use string_interner::backend::StringBackend;
+use string_interner::{DefaultSymbol, StringInterner};
+
+pub mod direct_import_queries;
+pub mod graph_manipulation;
+pub mod hierarchy_queries;
+pub mod higher_order_queries;
+pub mod import_chain_queries;
+
+pub(crate) mod pathfinding;
+
+lazy_static! {
+    static ref MODULE_NAMES: RwLock<StringInterner<StringBackend>> =
+        RwLock::new(StringInterner::default());
+    static ref EMPTY_MODULE_TOKENS: FxHashSet<ModuleToken> = FxHashSet::default();
+    static ref EMPTY_IMPORT_DETAILS: FxHashSet<ImportDetails> = FxHashSet::default();
+    static ref EMPTY_IMPORTS: FxHashSet<(ModuleToken, ModuleToken)> = FxHashSet::default();
```

Comment:
🤔 Rather than modelling imports as two-tuples of `ModuleToken`s, might be better to use a struct so it's more explicit which is the importer and which the imported. What do you think?

Reply:
No strong feeling, but your suggestion is probably a little nicer 👍